src/tir/transforms/lower_cross_thread_reduction.cc

```diff
@@ -478,6 +497,30 @@ class CrossThreadReductionTransformer : public StmtMutator {
     return need ? reduction_loops : std::vector<const ForNode*>{};
   }
 
+  // Check if the input block needs thread broadcast rewrite.
+  // One block needs broadcast rewrite when there exists one or more thread
+  // vars which vars free variables to this block.
+  std::vector<std::pair<ThreadScope, Range>> NeedCrossThreadBroadcast(
+      const BlockRealizeNode* realize) {
+    std::unordered_map<ThreadScope, Range, ThreadScopeHash, ThreadScopeEqual> unbound_thread2range =
+        thread2range_;
+    for (const PrimExpr& iter_value : realize->iter_values) {
```

Comment:
Checking iter_values have a few disadvantages, for example, unit loop could be unbound. Instead, check the surrounding `loop_stack_` which is much more robust

Reply:
Thanks for the good catch!! Updated.